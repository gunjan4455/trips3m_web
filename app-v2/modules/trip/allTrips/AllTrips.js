import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { GTripNavDiv } from './G';
import { ORGANISM_VIEWED } from 'actions/segmentEvents';
import { getTripsByBookingStatusLabel } from 'modules/trip/reducers/tripList';
import WhyTT from 'modules/rlp/whyTT/index';
import NoTrips from './components/NoTrips';
import TripCard from './components/TripCard';
import NavBar from './NavBar';

const TABS = {
  ACTIVE: 'active',
  INCOMPLETE: 'incomplete',
  BOOKED: 'booked'
};

class AllTrips extends Component {

  state = {
    tab: 'active'
  };

  componentDidMount() {
    this.props.trackSegment({
      event: ORGANISM_VIEWED,
      section: 'Active Requests'
    });
  }

  renderTrips(allTrips) {
    let tripIds = [];

    if (this.state.tab === 'active') {
      tripIds = getTripsByBookingStatusLabel(allTrips, TABS.ACTIVE);
    }
    else if (this.state.tab === TABS.INCOMPLETE) {
      tripIds = getTripsByBookingStatusLabel(allTrips, TABS.INCOMPLETE);
    }
    else if (this.state.tab === TABS.BOOKED) {
      tripIds = getTripsByBookingStatusLabel(allTrips, TABS.BOOKED);
    }

    if (!tripIds.length) {
      return <div>
        <NoTrips trackSegment={this.props.trackSegment} />
        <WhyTT.container />
      </div>;
    }

    return tripIds.map(id => <TripCard key={id} tripDetails={allTrips.byId[id]}
                                       reactivateBooking={this.props.reactivateBooking}
                                       trackSegment={this.props.trackSegment} />);
  }

  toggleTab = (tab) => {
    this.setState({ tab }, () => {
      this.props.trackSegment({
        event: ORGANISM_VIEWED,
        section: tab === TABS.ACTIVE ? 'Active Requests' : 'Incomplete Requests'
      });
    });
  };

  goBackHandler = () => {
    window.history.back();

    // currently we are coming on this page after refresh, This code will be used when we redirect here with react routing.
    // const { location, goBack } = this.props;
    // location.action === 'POP' ? this.props.push('/') : goBack();
  };

  render() {
    const { trips, trackSegment } = this.props;

    return (
      <div>
        <NavBar goBack={this.goBackHandler} />
        <div className="clearfix contentBackground sbc5">
          <div className="flex sbcw mb8 bb bs6 at_tripNav overflowXscroll">
            <GTripNavDiv
              className={`flexFull pt15 pb15 pl24 pr24 text-center relative nowrap ${this.state.tab === TABS.ACTIVE ? 'active' : null} at_active`}>
              <span className="f14" onClick={() => this.toggleTab(TABS.ACTIVE)}>Active Requests</span>
            </GTripNavDiv>
            <GTripNavDiv
              className={`flexFull pt15 pb15 pl24 pr24 text-center relative nowrap ${this.state.tab === TABS.INCOMPLETE ? 'active' : null} at_incomplete`}>
              <span className="f14 at_incompleteReq"
                    onClick={() => this.toggleTab(TABS.INCOMPLETE)}>Incomplete Requests</span>
            </GTripNavDiv>
            <GTripNavDiv
              className={`flexFull pt15 pb15 pl24 pr24 text-center relative nowrap ${this.state.tab === TABS.BOOKED ? 'active' : null} at_booked`}>
              <span className="f14 at_bookedReq"
                    onClick={() => this.toggleTab(TABS.BOOKED)}>Booked</span>
            </GTripNavDiv>
          </div>
          {trips.allIds.length ?
            this.renderTrips(trips) :
            <div>
              <NoTrips trackSegment={trackSegment} />
              <WhyTT.container />
            </div>
          }
        </div>
      </div>
    );
  }
}

AllTrips.propTypes = {
  trips: PropTypes.object.isRequired,
  location: PropTypes.object,
  reactivateBooking: PropTypes.func.isRequired,
  goBack: PropTypes.func.isRequired,
  trackSegment: PropTypes.func
};

AllTrips.defaultProps = {
  trackSegment: () => {}
};

export default AllTrips;
