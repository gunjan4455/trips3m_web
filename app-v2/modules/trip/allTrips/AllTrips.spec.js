import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import AllTrips from './AllTrips';
import TripCard from './components/TripCard';
import NoTrips from './components/NoTrips';

configure({ adapter: new Adapter() });

const DEFAULT_PROPS = {
  trips: {
    loaded: true,
    loading: false,
    allIds: [
      '5282142',
      '5282253',
      '5282255'
    ],
    byId: {
      '5282142': {
        id: 5282142,
        tripDays: 4,
        toLoc: 'Goa',
        firstDestinationName: 'Goa',
        startDate: '2019-12-04',
        quotesCount: 2,
        bookingStatusLabel: 'Incomplete',
        exclusivePreferencesValues: {
          cabs: 'Excluded',
          flights: 'Included',
          hotels: '5 Star'
        },
        budget: '21000..25000',
        fromLoc: 'Gurugram, Haryana, India',
        info: 'thyreyry',
        adult: 2,
        status: 'Cancelled',
        child: 1,
        tripStage: 3,
        currency: null,
        phoneNo: '+919876328752'
      },
      '5282253': {
        id: 5282253,
        tripDays: 3,
        toLoc: 'Goa',
        firstDestinationName: 'Goa',
        startDate: '2019-07-25',
        quotesCount: 0,
        bookingStatusLabel: 'Active',
        exclusivePreferencesValues: {
          cabs: 'Excluded',
          flights: 'Included',
          hotels: '5 Star'
        },
        budget: '17500..20500',
        fromLoc: 'Sarhol, Haryana 122022, India',
        info: 'uhoi',
        adult: 2,
        status: 'Active',
        child: 0,
        tripStage: 3,
        currency: null,
        phoneNo: '+917865876868'
      },
      '5282255': {
        id: 5282255,
        tripDays: 3,
        toLoc: 'Goa',
        firstDestinationName: 'Goa',
        startDate: '2019-07-25',
        quotesCount: 0,
        bookingStatusLabel: 'Incomplete',
        exclusivePreferencesValues: {
          cabs: 'Included',
          flights: 'Included',
          hotels: '5 Star'
        },
        budget: '18000..21000',
        fromLoc: 'Gurugram, Haryana, India',
        info: '',
        adult: 3,
        status: 'Dormant',
        child: 0,
        tripStage: 4,
        currency: null,
        phoneNo: '+918765432365'
      }
    }
  },
  location: {},
  reactivateBooking: jest.fn,
  push: jest.fn,
  goBack: jest.fn
};

const shallowWrapperAllTrips = (props) => {
  return shallow(<AllTrips {...props} />);
};

describe('Traveller dashboard module', () => {
  let props = {
    ...DEFAULT_PROPS,
    trips: { ...DEFAULT_PROPS.trips, allIds: [...DEFAULT_PROPS.trips.allIds], byId: { ...DEFAULT_PROPS.trips.byId } }
  };
  afterEach(() => {
    props = {
      ...DEFAULT_PROPS,
      trips: { ...DEFAULT_PROPS.trips, allIds: [...DEFAULT_PROPS.trips.allIds], byId: { ...DEFAULT_PROPS.trips.byId } }
    };
  });

  it('should render correctly', () => {
    const wrapper = shallowWrapperAllTrips(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('Incomplete Trips tab should have active class', () => {
    const wrapper = shallowWrapperAllTrips(props);
    wrapper.find('.at_incompleteReq').simulate('click');
    expect(wrapper.find('.at_incomplete').hasClass('active')).toEqual(true);
  });

  it('Should render 1 active trip and 2 incomplete trip', () => {
    const wrapper = shallowWrapperAllTrips(props);
    expect(wrapper.find(TripCard)).toHaveLength(1);

    wrapper.find('.at_incompleteReq').simulate('click');
    expect(wrapper.find(TripCard)).toHaveLength(2);
  });

  it('Should render NoTrips card when No trips is there', () => {
    props.trips.allIds = [];
    props.trips.byId = {};

    const wrapper = shallowWrapperAllTrips(props);
    expect(wrapper.find(NoTrips)).toHaveLength(1);
  });

  it('Should render NoTrips card when No active trips is there', () => {
    props.trips.allIds = [
      '5282142',
      '5282255'
    ];

    const wrapper = shallowWrapperAllTrips(props);
    expect(wrapper.find(NoTrips)).toHaveLength(1);

  });
});
