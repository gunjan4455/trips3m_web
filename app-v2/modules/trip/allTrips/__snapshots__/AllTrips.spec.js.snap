// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Traveller dashboard module should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AllTrips
    goBack={[Function]}
    location={Object {}}
    push={[Function]}
    reactivateBooking={[Function]}
    trackSegment={[Function]}
    trips={
      Object {
        "allIds": Array [
          "5282142",
          "5282253",
          "5282255",
        ],
        "byId": Object {
          "5282142": Object {
            "adult": 2,
            "bookingStatusLabel": "Incomplete",
            "budget": "21000..25000",
            "child": 1,
            "currency": null,
            "exclusivePreferencesValues": Object {
              "cabs": "Excluded",
              "flights": "Included",
              "hotels": "5 Star",
            },
            "firstDestinationName": "Goa",
            "fromLoc": "Gurugram, Haryana, India",
            "id": 5282142,
            "info": "thyreyry",
            "phoneNo": "+919876328752",
            "quotesCount": 2,
            "startDate": "2019-12-04",
            "status": "Cancelled",
            "toLoc": "Goa",
            "tripDays": 4,
            "tripStage": 3,
          },
          "5282253": Object {
            "adult": 2,
            "bookingStatusLabel": "Active",
            "budget": "17500..20500",
            "child": 0,
            "currency": null,
            "exclusivePreferencesValues": Object {
              "cabs": "Excluded",
              "flights": "Included",
              "hotels": "5 Star",
            },
            "firstDestinationName": "Goa",
            "fromLoc": "Sarhol, Haryana 122022, India",
            "id": 5282253,
            "info": "uhoi",
            "phoneNo": "+917865876868",
            "quotesCount": 0,
            "startDate": "2019-07-25",
            "status": "Active",
            "toLoc": "Goa",
            "tripDays": 3,
            "tripStage": 3,
          },
          "5282255": Object {
            "adult": 3,
            "bookingStatusLabel": "Incomplete",
            "budget": "18000..21000",
            "child": 0,
            "currency": null,
            "exclusivePreferencesValues": Object {
              "cabs": "Included",
              "flights": "Included",
              "hotels": "5 Star",
            },
            "firstDestinationName": "Goa",
            "fromLoc": "Gurugram, Haryana, India",
            "id": 5282255,
            "info": "",
            "phoneNo": "+918765432365",
            "quotesCount": 0,
            "startDate": "2019-07-25",
            "status": "Dormant",
            "toLoc": "Goa",
            "tripDays": 3,
            "tripStage": 4,
          },
        },
        "loaded": true,
        "loading": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavBar
          goBack={[Function]}
          trackSegment={[Function]}
        />,
        <div
          className="clearfix contentBackground sbc5"
        >
          <div
            className="flex sbcw mb8 bb bs6 at_tripNav overflowXscroll"
          >
            <ForwardRef(render)
              className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap active at_active"
            >
              <span
                className="f14"
                onClick={[Function]}
              >
                Active Requests
              </span>
            </ForwardRef(render)>
            <ForwardRef(render)
              className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap null at_incomplete"
            >
              <span
                className="f14 at_incompleteReq"
                onClick={[Function]}
              >
                Incomplete Requests
              </span>
            </ForwardRef(render)>
            <ForwardRef(render)
              className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap null at_booked"
            >
              <span
                className="f14 at_bookedReq"
                onClick={[Function]}
              >
                Booked
              </span>
            </ForwardRef(render)>
          </div>
          <TripCard
            reactivateBooking={[Function]}
            trackSegment={[Function]}
            tripDetails={
              Object {
                "adult": 2,
                "bookingStatusLabel": "Active",
                "budget": "17500..20500",
                "child": 0,
                "currency": null,
                "exclusivePreferencesValues": Object {
                  "cabs": "Excluded",
                  "flights": "Included",
                  "hotels": "5 Star",
                },
                "firstDestinationName": "Goa",
                "fromLoc": "Sarhol, Haryana 122022, India",
                "id": 5282253,
                "info": "uhoi",
                "phoneNo": "+917865876868",
                "quotesCount": 0,
                "startDate": "2019-07-25",
                "status": "Active",
                "toLoc": "Goa",
                "tripDays": 3,
                "tripStage": 3,
              }
            }
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "goBack": [Function],
          "trackSegment": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="flex sbcw mb8 bb bs6 at_tripNav overflowXscroll"
            >
              <ForwardRef(render)
                className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap active at_active"
              >
                <span
                  className="f14"
                  onClick={[Function]}
                >
                  Active Requests
                </span>
              </ForwardRef(render)>
              <ForwardRef(render)
                className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap null at_incomplete"
              >
                <span
                  className="f14 at_incompleteReq"
                  onClick={[Function]}
                >
                  Incomplete Requests
                </span>
              </ForwardRef(render)>
              <ForwardRef(render)
                className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap null at_booked"
              >
                <span
                  className="f14 at_bookedReq"
                  onClick={[Function]}
                >
                  Booked
                </span>
              </ForwardRef(render)>
            </div>,
            Array [
              <TripCard
                reactivateBooking={[Function]}
                trackSegment={[Function]}
                tripDetails={
                  Object {
                    "adult": 2,
                    "bookingStatusLabel": "Active",
                    "budget": "17500..20500",
                    "child": 0,
                    "currency": null,
                    "exclusivePreferencesValues": Object {
                      "cabs": "Excluded",
                      "flights": "Included",
                      "hotels": "5 Star",
                    },
                    "firstDestinationName": "Goa",
                    "fromLoc": "Sarhol, Haryana 122022, India",
                    "id": 5282253,
                    "info": "uhoi",
                    "phoneNo": "+917865876868",
                    "quotesCount": 0,
                    "startDate": "2019-07-25",
                    "status": "Active",
                    "toLoc": "Goa",
                    "tripDays": 3,
                    "tripStage": 3,
                  }
                }
              />,
            ],
          ],
          "className": "clearfix contentBackground sbc5",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <ForwardRef(render)
                  className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap active at_active"
                >
                  <span
                    className="f14"
                    onClick={[Function]}
                  >
                    Active Requests
                  </span>
                </ForwardRef(render)>,
                <ForwardRef(render)
                  className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap null at_incomplete"
                >
                  <span
                    className="f14 at_incompleteReq"
                    onClick={[Function]}
                  >
                    Incomplete Requests
                  </span>
                </ForwardRef(render)>,
                <ForwardRef(render)
                  className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap null at_booked"
                >
                  <span
                    className="f14 at_bookedReq"
                    onClick={[Function]}
                  >
                    Booked
                  </span>
                </ForwardRef(render)>,
              ],
              "className": "flex sbcw mb8 bb bs6 at_tripNav overflowXscroll",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <span
                    className="f14"
                    onClick={[Function]}
                  >
                    Active Requests
                  </span>,
                  "className": "flexFull pt15 pb15 pl24 pr24 text-center relative nowrap active at_active",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Active Requests",
                    "className": "f14",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Active Requests",
                  "type": "span",
                },
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "__emotion_base": "div",
                  "__emotion_forwardProp": undefined,
                  "__emotion_real": [Circular],
                  "__emotion_styles": Array [
                    Object {
                      "&.active": Object {
                        "&:before": Object {
                          "background": "#20a397",
                          "bottom": "-1px",
                          "content": "\\"\\"",
                          "height": "3px;",
                          "left": "0",
                          "position": "absolute",
                          "right": "0",
                        },
                        "color": "#20a397",
                      },
                      "cursor": "pointer",
                    },
                  ],
                  "defaultProps": undefined,
                  "displayName": "Styled(div)",
                  "render": [Function],
                  "withComponent": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <span
                    className="f14 at_incompleteReq"
                    onClick={[Function]}
                  >
                    Incomplete Requests
                  </span>,
                  "className": "flexFull pt15 pb15 pl24 pr24 text-center relative nowrap null at_incomplete",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Incomplete Requests",
                    "className": "f14 at_incompleteReq",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Incomplete Requests",
                  "type": "span",
                },
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "__emotion_base": "div",
                  "__emotion_forwardProp": undefined,
                  "__emotion_real": [Circular],
                  "__emotion_styles": Array [
                    Object {
                      "&.active": Object {
                        "&:before": Object {
                          "background": "#20a397",
                          "bottom": "-1px",
                          "content": "\\"\\"",
                          "height": "3px;",
                          "left": "0",
                          "position": "absolute",
                          "right": "0",
                        },
                        "color": "#20a397",
                      },
                      "cursor": "pointer",
                    },
                  ],
                  "defaultProps": undefined,
                  "displayName": "Styled(div)",
                  "render": [Function],
                  "withComponent": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <span
                    className="f14 at_bookedReq"
                    onClick={[Function]}
                  >
                    Booked
                  </span>,
                  "className": "flexFull pt15 pb15 pl24 pr24 text-center relative nowrap null at_booked",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Booked",
                    "className": "f14 at_bookedReq",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Booked",
                  "type": "span",
                },
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "__emotion_base": "div",
                  "__emotion_forwardProp": undefined,
                  "__emotion_real": [Circular],
                  "__emotion_styles": Array [
                    Object {
                      "&.active": Object {
                        "&:before": Object {
                          "background": "#20a397",
                          "bottom": "-1px",
                          "content": "\\"\\"",
                          "height": "3px;",
                          "left": "0",
                          "position": "absolute",
                          "right": "0",
                        },
                        "color": "#20a397",
                      },
                      "cursor": "pointer",
                    },
                  ],
                  "defaultProps": undefined,
                  "displayName": "Styled(div)",
                  "render": [Function],
                  "withComponent": [Function],
                },
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "5282253",
            "nodeType": "class",
            "props": Object {
              "reactivateBooking": [Function],
              "trackSegment": [Function],
              "tripDetails": Object {
                "adult": 2,
                "bookingStatusLabel": "Active",
                "budget": "17500..20500",
                "child": 0,
                "currency": null,
                "exclusivePreferencesValues": Object {
                  "cabs": "Excluded",
                  "flights": "Included",
                  "hotels": "5 Star",
                },
                "firstDestinationName": "Goa",
                "fromLoc": "Sarhol, Haryana 122022, India",
                "id": 5282253,
                "info": "uhoi",
                "phoneNo": "+917865876868",
                "quotesCount": 0,
                "startDate": "2019-07-25",
                "status": "Active",
                "toLoc": "Goa",
                "tripDays": 3,
                "tripStage": 3,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavBar
            goBack={[Function]}
            trackSegment={[Function]}
          />,
          <div
            className="clearfix contentBackground sbc5"
          >
            <div
              className="flex sbcw mb8 bb bs6 at_tripNav overflowXscroll"
            >
              <ForwardRef(render)
                className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap active at_active"
              >
                <span
                  className="f14"
                  onClick={[Function]}
                >
                  Active Requests
                </span>
              </ForwardRef(render)>
              <ForwardRef(render)
                className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap null at_incomplete"
              >
                <span
                  className="f14 at_incompleteReq"
                  onClick={[Function]}
                >
                  Incomplete Requests
                </span>
              </ForwardRef(render)>
              <ForwardRef(render)
                className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap null at_booked"
              >
                <span
                  className="f14 at_bookedReq"
                  onClick={[Function]}
                >
                  Booked
                </span>
              </ForwardRef(render)>
            </div>
            <TripCard
              reactivateBooking={[Function]}
              trackSegment={[Function]}
              tripDetails={
                Object {
                  "adult": 2,
                  "bookingStatusLabel": "Active",
                  "budget": "17500..20500",
                  "child": 0,
                  "currency": null,
                  "exclusivePreferencesValues": Object {
                    "cabs": "Excluded",
                    "flights": "Included",
                    "hotels": "5 Star",
                  },
                  "firstDestinationName": "Goa",
                  "fromLoc": "Sarhol, Haryana 122022, India",
                  "id": 5282253,
                  "info": "uhoi",
                  "phoneNo": "+917865876868",
                  "quotesCount": 0,
                  "startDate": "2019-07-25",
                  "status": "Active",
                  "toLoc": "Goa",
                  "tripDays": 3,
                  "tripStage": 3,
                }
              }
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "goBack": [Function],
            "trackSegment": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="flex sbcw mb8 bb bs6 at_tripNav overflowXscroll"
              >
                <ForwardRef(render)
                  className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap active at_active"
                >
                  <span
                    className="f14"
                    onClick={[Function]}
                  >
                    Active Requests
                  </span>
                </ForwardRef(render)>
                <ForwardRef(render)
                  className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap null at_incomplete"
                >
                  <span
                    className="f14 at_incompleteReq"
                    onClick={[Function]}
                  >
                    Incomplete Requests
                  </span>
                </ForwardRef(render)>
                <ForwardRef(render)
                  className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap null at_booked"
                >
                  <span
                    className="f14 at_bookedReq"
                    onClick={[Function]}
                  >
                    Booked
                  </span>
                </ForwardRef(render)>
              </div>,
              Array [
                <TripCard
                  reactivateBooking={[Function]}
                  trackSegment={[Function]}
                  tripDetails={
                    Object {
                      "adult": 2,
                      "bookingStatusLabel": "Active",
                      "budget": "17500..20500",
                      "child": 0,
                      "currency": null,
                      "exclusivePreferencesValues": Object {
                        "cabs": "Excluded",
                        "flights": "Included",
                        "hotels": "5 Star",
                      },
                      "firstDestinationName": "Goa",
                      "fromLoc": "Sarhol, Haryana 122022, India",
                      "id": 5282253,
                      "info": "uhoi",
                      "phoneNo": "+917865876868",
                      "quotesCount": 0,
                      "startDate": "2019-07-25",
                      "status": "Active",
                      "toLoc": "Goa",
                      "tripDays": 3,
                      "tripStage": 3,
                    }
                  }
                />,
              ],
            ],
            "className": "clearfix contentBackground sbc5",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <ForwardRef(render)
                    className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap active at_active"
                  >
                    <span
                      className="f14"
                      onClick={[Function]}
                    >
                      Active Requests
                    </span>
                  </ForwardRef(render)>,
                  <ForwardRef(render)
                    className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap null at_incomplete"
                  >
                    <span
                      className="f14 at_incompleteReq"
                      onClick={[Function]}
                    >
                      Incomplete Requests
                    </span>
                  </ForwardRef(render)>,
                  <ForwardRef(render)
                    className="flexFull pt15 pb15 pl24 pr24 text-center relative nowrap null at_booked"
                  >
                    <span
                      className="f14 at_bookedReq"
                      onClick={[Function]}
                    >
                      Booked
                    </span>
                  </ForwardRef(render)>,
                ],
                "className": "flex sbcw mb8 bb bs6 at_tripNav overflowXscroll",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <span
                      className="f14"
                      onClick={[Function]}
                    >
                      Active Requests
                    </span>,
                    "className": "flexFull pt15 pb15 pl24 pr24 text-center relative nowrap active at_active",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Active Requests",
                      "className": "f14",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Active Requests",
                    "type": "span",
                  },
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "__emotion_base": "div",
                    "__emotion_forwardProp": undefined,
                    "__emotion_real": [Circular],
                    "__emotion_styles": Array [
                      Object {
                        "&.active": Object {
                          "&:before": Object {
                            "background": "#20a397",
                            "bottom": "-1px",
                            "content": "\\"\\"",
                            "height": "3px;",
                            "left": "0",
                            "position": "absolute",
                            "right": "0",
                          },
                          "color": "#20a397",
                        },
                        "cursor": "pointer",
                      },
                    ],
                    "defaultProps": undefined,
                    "displayName": "Styled(div)",
                    "render": [Function],
                    "withComponent": [Function],
                  },
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <span
                      className="f14 at_incompleteReq"
                      onClick={[Function]}
                    >
                      Incomplete Requests
                    </span>,
                    "className": "flexFull pt15 pb15 pl24 pr24 text-center relative nowrap null at_incomplete",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Incomplete Requests",
                      "className": "f14 at_incompleteReq",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Incomplete Requests",
                    "type": "span",
                  },
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "__emotion_base": "div",
                    "__emotion_forwardProp": undefined,
                    "__emotion_real": [Circular],
                    "__emotion_styles": Array [
                      Object {
                        "&.active": Object {
                          "&:before": Object {
                            "background": "#20a397",
                            "bottom": "-1px",
                            "content": "\\"\\"",
                            "height": "3px;",
                            "left": "0",
                            "position": "absolute",
                            "right": "0",
                          },
                          "color": "#20a397",
                        },
                        "cursor": "pointer",
                      },
                    ],
                    "defaultProps": undefined,
                    "displayName": "Styled(div)",
                    "render": [Function],
                    "withComponent": [Function],
                  },
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <span
                      className="f14 at_bookedReq"
                      onClick={[Function]}
                    >
                      Booked
                    </span>,
                    "className": "flexFull pt15 pb15 pl24 pr24 text-center relative nowrap null at_booked",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Booked",
                      "className": "f14 at_bookedReq",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Booked",
                    "type": "span",
                  },
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "__emotion_base": "div",
                    "__emotion_forwardProp": undefined,
                    "__emotion_real": [Circular],
                    "__emotion_styles": Array [
                      Object {
                        "&.active": Object {
                          "&:before": Object {
                            "background": "#20a397",
                            "bottom": "-1px",
                            "content": "\\"\\"",
                            "height": "3px;",
                            "left": "0",
                            "position": "absolute",
                            "right": "0",
                          },
                          "color": "#20a397",
                        },
                        "cursor": "pointer",
                      },
                    ],
                    "defaultProps": undefined,
                    "displayName": "Styled(div)",
                    "render": [Function],
                    "withComponent": [Function],
                  },
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "5282253",
              "nodeType": "class",
              "props": Object {
                "reactivateBooking": [Function],
                "trackSegment": [Function],
                "tripDetails": Object {
                  "adult": 2,
                  "bookingStatusLabel": "Active",
                  "budget": "17500..20500",
                  "child": 0,
                  "currency": null,
                  "exclusivePreferencesValues": Object {
                    "cabs": "Excluded",
                    "flights": "Included",
                    "hotels": "5 Star",
                  },
                  "firstDestinationName": "Goa",
                  "fromLoc": "Sarhol, Haryana 122022, India",
                  "id": 5282253,
                  "info": "uhoi",
                  "phoneNo": "+917865876868",
                  "quotesCount": 0,
                  "startDate": "2019-07-25",
                  "status": "Active",
                  "toLoc": "Goa",
                  "tripDays": 3,
                  "tripStage": 3,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
